integration/spark/src/main/scala/org/apache/spark/sql/index/CarbonIndexUtil.scala

```diff
@@ -377,4 +381,212 @@ object CarbonIndexUtil {
       AlterTableUtil.releaseLocks(locks.asScala.toList)
     }
   }
+
+  def processSIRepair(indexTableName: String, carbonTable: CarbonTable,
+    carbonLoadModel: CarbonLoadModel, indexMetadata: IndexMetadata,
+      mainTableDetails: List[LoadMetadataDetails], secondaryIndexProvider: String)
+  (sparkSession: SparkSession) : Unit = {
+    val sparkSession = SparkSession.getActiveSession.get
+    // val databaseName = sparkSession.catalog.currentDatabase
+    // when Si creation and load to main table are parallel, get the carbonTable from the
+    // metastore which will have the latest index Info
+    val metaStore = CarbonEnv.getInstance(sparkSession).carbonMetaStore
+    val indexTable = metaStore
+      .lookupRelation(Some(carbonLoadModel.getDatabaseName), indexTableName)(
+        sparkSession)
+      .asInstanceOf[CarbonRelation]
+      .carbonTable
+
+    val mainTblLoadMetadataDetails: Array[LoadMetadataDetails] =
+      SegmentStatusManager.readLoadMetadata(carbonTable.getMetadataPath)
```

Comment:
consider passing the loadMetadata details from the caller to avoid multiple reading.

Reply:
removed multiple readings, reading only once

Reply:
Why "readLoadMetadata" is till there for maintable?

Reply:
added it from the caller